atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraNamespaceDeleterTest.java

```diff
@@ -0,0 +1,186 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class CassandraNamespaceDeleterTest {
+    private static final String KEYSPACE = "nonreservedkeyspace";
+    private static final ImmutableCqlQuery DROP_KEYSPACE_QUERY = CqlQuery.builder()
+            .safeQueryFormat(SchemaBuilder.dropKeyspace("\"" + KEYSPACE + "\"")
+                    .ifExists()
+                    .buildInternal())
+            .build();
+
+    @Mock
+    private CassandraClient cassandraClient;
+
+    private NamespaceDeleter namespaceDeleter;
+
+    @Before
+    public void before() {
+        namespaceDeleter = new CassandraNamespaceDeleter(getConfig(), () -> cassandraClient);
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceDropsKeyspaceSpecifiedInConfig() throws TException {
+        setupCassandraSchemaVersions();
+
+        namespaceDeleter.deleteAllDataFromNamespace();
+
+        verify(cassandraClient).execute_cql3_query(eq(DROP_KEYSPACE_QUERY), any(), any());
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceDropsOnlyKeyspaceSpecified() throws TException {
+        setupCassandraSchemaVersions();
+
+        namespaceDeleter.deleteAllDataFromNamespace();
+
+        verify(cassandraClient, times(1)).execute_cql3_query(any(), any(), any());
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceWaitsForSchemaVersionsBeforeDroppingKeyspace() throws TException {
+        when(cassandraClient.describe_schema_versions()).thenReturn(Map.of("hello", List.of(), "world", List.of()));
+
+        assertThatThrownBy(namespaceDeleter::deleteAllDataFromNamespace)
+                .hasCauseInstanceOf(IllegalStateException.class)
+                .cause()
+                .hasMessageContaining("Cassandra cluster cannot come to agreement on schema versions");
+        verify(cassandraClient, never()).execute_cql3_query(any(), any(), any());
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceWaitsForSchemaVersionsAfterDroppingKeyspace() throws TException {
+        when(cassandraClient.describe_schema_versions())
+                .thenReturn(Map.of("hello", List.of("world")))
+                .thenReturn(Map.of("hello", List.of(), "world", List.of()));
+
+        assertThatThrownBy(namespaceDeleter::deleteAllDataFromNamespace)
+                .hasCauseInstanceOf(IllegalStateException.class)
+                .cause()
+                .hasMessageContaining("Cassandra cluster cannot come to agreement on schema versions");
+
+        verify(cassandraClient).execute_cql3_query(eq(DROP_KEYSPACE_QUERY), any(), any());
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceClosesClient() throws TException {
+        setupCassandraSchemaVersions();
+
+        namespaceDeleter.deleteAllDataFromNamespace();
+        verify(cassandraClient).close();
+    }
```

Comment:
like we have for `isNamespaceDeletedSuccessfully`, we probably want tests for exceptional cases